src/Application.php

```diff
@@ -9,14 +9,70 @@
 
 class Application extends  SymfonyApplication
 {
+    /**
+     * @var \Robo\TaskAssembler
+     */
+    protected $taskAssembler;
+
+    public function __construct($name, $version)
+    {
+        parent::__construct($name, $version);
+
+        $this->getDefinition()->addOption(
+            new InputOption('--simulate', null, InputOption::VALUE_NONE, 'Run in simulated mode (show what would have happened).')
+        );
+    }
+
+    public function setTaskAssembler($taskAssembler)
+    {
+        $this->taskAssembler = $taskAssembler;
+    }
+
+    public function taskAssembler()
+    {
+        return $this->taskAssembler;
+    }
+
+    protected static function findAccessorMethods($commandNames)
+    {
+        $accessorNames = [];
+        // Using get_class_vars is not reliable, because it only
+        // exposes public class variables.
+        foreach ($commandNames as $commandName) {
+            // We count a set of methods as accessor methods if
+            // there is a setter and at least one getter.
+            if (strpos($commandName, 'set') === 0) {
+                $name = lcfirst(substr($commandName, 3));
+                $getter = 'get' . ucfirst($name);
+                $hasGetter = false;
+                if (in_array($name, $commandNames)) {
+                    $accessorNames[] = $name;
+                    $hasGetter = true;
+                }
+                if (in_array($getter, $commandNames)) {
+                    $accessorNames[] = $getter;
+                    $hasGetter = true;
+                }
+                if ($hasGetter) {
+                    $accessorNames[] = $commandName;
+                }
+            }
+        }
+        return $accessorNames;
+    }
 
     public function addCommandsFromClass($className, $passThrough = null)
     {
         $roboTasks = new $className;
+        if ($roboTasks instanceof \Robo\Tasks) {
+            $roboTasks->setTaskAssembler($this->taskAssembler);
+        }
 
-        $commandNames = array_filter(get_class_methods($className), function($m) {
+        $commandNames = array_filter(get_class_methods($className), function ($m) {
             return !in_array($m, ['__construct']);
         });
+        $accessorNames = static::findAccessorMethods($commandNames);
+        $commandNames = array_diff($commandNames, $accessorNames);
```

Comment:
I'm not sure why you need `findAccessorMethods` method actually. 
If you need to ignore getters in setters in command list, just ignore commands starting from `get` and `set` prefexes (but take camelCase into account `setUp` - ignored, but `setup` added as command)